common/thorhelper/thorsoapcall.cpp

```diff
@@ -941,46 +938,102 @@ class CWSCHelper : implements IWSCHelper, public CInterface
         else
             rowTransformer = NULL;
 
+        StringBuffer proxyAddress;
         OwnedRoxieString hosts(helper->getHosts());
-        UrlListParser urlListParser(hosts);
-        if ((numUrls = urlListParser.getUrls(urlArray)) > 0)
+        if (isEmptyString(hosts))
+            throw MakeStringException(0, "%sCALL specified no URLs",wscType == STsoap ? "SOAP" : "HTTP");
+        if (0==strncmp(hosts, "secret:", 7))
         {
-            if (wscMode == SCrow)
+            const char *finger = hosts.get()+7;
+            if (isEmptyString(finger))
+                throw MakeStringException(0, "%sCALL HTTP-CONNECT SECRET specified with no name", wscType == STsoap ? "SOAP" : "HTTP");
+
+            StringBuffer secretName("http-connect-");
+            secretName.append(finger);
+            if (!isEmptyString(helper->getProxyAddress()))
```

Comment:
This will potentially leak - see the previous code which assigned it to a variable first.